operator/pkg/preflight/configconnectorcontextchecker.go

```diff
@@ -52,5 +53,21 @@ func (c *ConfigConnectorContextChecker) Preflight(_ context.Context, o declarati
 		return fmt.Errorf("spec.billingProject must be set if spec.requestProjectPolicy is set to %v", k8s.BillingProjectPolicy)
 	}
 
+	if err := validateGSAFormat(ccc.Spec.GoogleServiceAccount); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateGSAFormat(gsa string) error {
+	if gsa == "" { // GoogleServiceAccount is a required field. We do not need to fail here.
+		return nil
+	}
+	validGSAPattern := `^[A-Za-z0-9._%+\-]+@[a-z0-9.\-]+\.gserviceaccount.com$`
```

Comment:
Where did you get this regex from?  I found https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts/create which suggested a more restrictive regex for the first bit.

projectID matches description here https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects

I'm actually more concerned about us rejecting _valid_ gcpServiceAccount values though, so as long as our regex is _too_ permissive, that's OK IMO!

Reply:
> Where did you get this regex from

I think I got the regex from a back-and-forth discussion between me and AI... 

> I found https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts/create which suggested a more restrictive regex for the first bit.

Good find! We could certainly make the first bit more restrictive to match Cloud IAM, but I feel leaving it less restrictive has more benefits.
 - If Cloud IAM update the regex in the future, our validation will likely still work.
 - IMO the primary goal is to reject really wrong formats, like if user forgot to replace "${GSA}" with the actual email address, it should get rejected here. (today we don't reject it, and annotate KSA with any string user write to this field)